fml/file.h

```diff
@@ -73,12 +78,37 @@ bool WriteAtomically(const fml::UniqueFD& base_directory,
                      const char* file_name,
                      const Mapping& mapping);
 
+using FileVisitor = std::function<void(const fml::UniqueFD& directory,
```

Comment:
Can we make the `FileVisitor` return a `bool` to indicate if further traversal is necessary? For example, folks can use this to implement a `FindFile` whereby they can return `false` when a file is found so further traversal is unnecessary.

Reply:
Please document if the filename is relative or absolute.

Reply:
`filename` inherently implies it's the name of the file as opposed to filepath, no? :-)